### PR TITLE
doc: Create pull_request_template.md

### PR DESCRIPTION
**Pull request by jcsp**
Let's take the opportunity to nudge new contributors
to have well formatted commit messages.

Signed-off-by: John Spray <john.spray@redhat.com>

### CODE REVIEW

**Comment by batrick** [comment]
I'd suggest also telling them that the ticket is especially important if they want their fix backported as that creates an incentive.

**Comment by jcsp** [comment]
Added the note about creating tickets for bugs, and moved into .github.

@tchaikov the HTML style comment is what I've seen in other examples -- it seems to work, not sure if it's a github specific hack for templates, or if it just works because the html comment marks get passed through.

**Comment by gregsfortytwo** [comment]
We shouldn't ask for a Signed-off-by in official documentation without linking to a statement about what it actually means. :) Rest looks good!

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+give the component as "mgr/<module name>" rather than a path into pybind.
+
+For more examples, simply use "git log" and look at some historical commits.
+
+If this is a bug fix, and a ticket does not yet exist on tracker.ceph.com,
+please create one now and reference it from your commit message.  Even for
+small or obvious fixes, the tracker ticket is important to enable
+the backporting process to stable branches.
+
+More information for contributors is available here:
```
Comment (LenzGr) [inline .github/pull_request_template.md]: I would move this reference further up, directly after the "Thank you!" line. Then I would start the following part with "In a nutshell, this is what you should do:"

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+https://raw.githubusercontent.com/ceph/ceph/master/SubmittingPatches.rst
+
+-->
+
+- [] References tracker ticket
```
[the remark this answers is not in the document]
Comment (tchaikov) [reply .github/pull_request_template.md]: i think it's needed.

[the remark this answers is not in the document]
Comment (smithfarm) [reply .github/pull_request_template.md]: yes, the space is needed - verified just now

**Comment by jcsp** [comment]
Updated to call out Signed-off-by's importance, add the space to the `[ ]` bits, and edit it down a bit.

@LenzGr I'm avoiding putting the SubmittingPatches.rst reference right at the top, because I want peoples eyeballs to gravitate to that example commit message -- once someone goes to SubmittingPatches I worry they get lost in the details, so I'm leaving that link til the end.

**Pull request merged**